concrete/src/Form/Service/DestinationPicker/PickerInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Concrete\Core\Form\Service\DestinationPicker;
+
+use ArrayAccess;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+interface PickerInterface
+{
+    /**
+     * Get the display name of this picker (to be used in the SELECT html element).
+     *
+     * @param array $options
+     *
+     * @return string
+     */
+    public function getDisplayName(array $options);
+
+    /**
+     * Get the height of this picker (in pixels).
+     *
+     * @return int
+     */
+    public function getHeight();
+
+    /**
+     * Generate the HTML for the picker.
+     *
+     * @param string $pickerKey
+     * @param array $options
+     * @param mixed|null $selectedValue
+     *
+     * @return string
+     */
+    public function generate($pickerKey, array $options, $selectedValue = null);
+
+    /**
+     * Decode the value received via post.
+     *
+     * @param \Symfony\Component\HttpFoundation\ParameterBag $data
+     * @param string $pickerKey
+     * @param array $options
+     * @param \ArrayAccess $errors
+     * @param string|null $fieldDisplayName
+     */
+    public function decode(ParameterBag $data, $pickerKey, array $options, ArrayAccess $errors = null, $fieldDisplayName = null);
```

Comment:
@mlocati shouldn't `$errors` be passed by reference?

Reply:
Wells, if you wanted the errors, pass an ArrayAccess instance. If you don't care about them, just pass null